src/coreclr/vm/interopconverter.cpp

```diff
@@ -189,17 +189,26 @@ IUnknown *GetComIPFromObjectRef(OBJECTREF *poref, ComIpType ReqIpType, ComIpType
         {
             hr = SafeQueryInterface(pUnk, IID_IDispatch, &pvObj);
             pUnk->Release();
+            if (!FAILED(hr))
+            {
+                FetchedIpType = ComIpType_Dispatch;
+            }
+            else
+            {
+                FetchedIpType = ComIpType_Unknown;
+            }
         }
         else
         {
             pvObj = pUnk;
+            FetchedIpType = ComIpType_Unknown;
         }
 
         if (FAILED(hr))
```

Comment:
The failure to QI for `IDispatch` shouldn't be a catastrophic failure &ndash; this was part of the bug I believe.